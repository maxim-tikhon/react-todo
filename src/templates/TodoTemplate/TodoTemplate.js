import TaskInput from "../../molecules/TaskInput/TaskInput";
import TaskItem from "../../molecules/TaskItem/TaskItem";
import TodoFooter from "../../molecules/TodoFooter/TodoFooter";
import Card from "../../organisms/Card/Card";
import styles from "./TodoTemplate.module.scss";

const TodoTemplate = ({ tasks, onAddTask }) => {
	const addNewTask = (description) => {
		const newTask = {
			id: Date(),
			description,
			completed: false
		};

		onAddTask(newTask);
	};

	return (
		<div className={styles.todoContainer}>
			<section className={styles.todoSection}>
				<header>
					<h1 className={styles.todoHeader}>TODO</h1>
				</header>

				<Card>
					<TaskInput onEnterTask={addNewTask} />
				</Card>

				<Card>
					{tasks.map((task) => (
						<TaskItem task={task} key={task.id} />
					))}
					<TodoFooter />
				</Card>

				<div className={styles.todoNote}>Drag and drop to reoder list</div>
			</section>
		</div>
	);
};

export default TodoTemplate;
