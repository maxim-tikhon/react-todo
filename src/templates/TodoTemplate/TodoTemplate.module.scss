.todoContainer {
	display: flex;
	flex-direction: column;
	align-items: center;
	background-image: linear-gradient(to bottom, #e983dd80, #579ebfcc),
		url("../../../public/images/header.jpg");
	height: 280px;
	background-size: cover;
	opacity: 0.9;

  .todoSection {
    width: 50vw;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    padding: 3rem 0;

    .todoHeader {
      color: #ffffff;
      font-size: 3rem;
      letter-spacing: 1rem;
      margin-bottom: 2rem;
    }

    .todoNote {
      color: grey;
      font-size: 14px;
      margin-top: 3rem;
      align-self: center;
    }
  }

  @media (max-width: 768px) {
    .todoSection {
      width: 90vw;
    }
  } 
}
