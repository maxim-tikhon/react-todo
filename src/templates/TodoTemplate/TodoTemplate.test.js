import renderer from 'react-test-renderer';
import TodoTemplate from './TodoTemplate';

describe('TodoTemplate', () => {
	it('renders correctly', () => {
		const output = renderer.create(<TodoTemplate />).toJSON();
		expect(output).toMatchSnapshot();
	});
});
