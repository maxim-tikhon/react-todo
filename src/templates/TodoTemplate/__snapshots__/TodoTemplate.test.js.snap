// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoTemplate renders correctly 1`] = `
<div
  className="todoContainer"
>
  <section
    className="todoSection"
  >
    <header>
      <h1
        className="todoHeader"
      >
        TODO
      </h1>
    </header>
    <div
      className="card single"
    >
      <input
        className="customCheckbox"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <input
        className="textInput newTaskInput"
        onKeyDown={[Function]}
        placeholder="Please enter a new task and click Enter to save"
        type="text"
      />
    </div>
  </section>
</div>
`;
